Assets/ModScripts/Submodules/WhosWho.cs

```diff
@@ -0,0 +1,293 @@
+using System.Collections.Generic;
+using System.Diagnostics; // to track how long each button has been held
+using System.Linq;
+using UnityEngine;
+using KModkit;
+using Random = UnityEngine.Random;
+
+public class WhosWho : Puzzle {
+    readonly int LEDtoUse;
+    readonly string LEDcolor;
+    readonly Dictionary<string, int> offsets = new Dictionary<string, int>()
+    {
+        {"red", 1},
+        {"orange", 2},
+        {"yellow", 3},
+        {"green", -3},
+        {"lime", -1},
+        {"cyan", -2},
+        {"blue", 4},
+        {"purple", 5},
+        {"pink", -5},
+        {"black", -4},
+        {"white", 6},
+    };
+    readonly Dictionary<string, int> rows = new Dictionary<string, int>()
+    {
+        {"red", 2},
+        {"orange", 1},
+        {"yellow", 1},
+        {"green", 3},
+        {"lime", 4},
+        {"cyan", 0},
+        {"blue", 5},
+        {"purple", 6},
+        {"pink", 2},
+        {"black", 0},
+        {"white", 7},
+    };
+
+    readonly Stopwatch O_PressTime = new Stopwatch();
+    readonly Stopwatch I_PressTime = new Stopwatch();
```

Comment:
Is having two Stopwatches necessary? You can't use press both buttons at the same time.